### PR TITLE
scope auth to webhooks and repos

### PR DESCRIPTION
**Pull request by MaxBittker**
this is not ready to merge yet, but will need to be merged in tandem with the bitbucket commit data plugin  

https://github.com/getsentry/sentry-plugins/pull/180

### CODE REVIEW

File: src/social_auth/backends/bitbucket.py [hunk 1]
```diff
@@ -76,6 +76,7 @@ class BitbucketAuth(ConsumerBasedOAuth):
     AUTH_BACKEND = BitbucketBackend
     SETTINGS_KEY_NAME = 'BITBUCKET_CONSUMER_KEY'
     SETTINGS_SECRET_NAME = 'BITBUCKET_CONSUMER_SECRET'
+    DEFAULT_SCOPE = ['webhook', 'repository', 'issue']
```
Comment (MaxBittker) [inline src/social_auth/backends/bitbucket.py]: I hope that existing bitbucket authed repos with old scopes don't have problems with this

File: src/sentry/api/endpoints/organization_config_repositories.py [hunk 2]
```diff
@@ -13,11 +13,12 @@ def get(self, request, organization):
         providers = []
         for provider_id in provider_bindings:
             provider = provider_bindings.get(provider_id)(id=provider_id)
-            providers.append({
-                'id': provider_id,
-                'name': provider.name,
-                'config': provider.get_config(),
-            })
+            if provider.name != 'Bitbucket' or bool(organization.flags.early_adopter):
```
Comment (macqueen) [inline src/sentry/api/endpoints/organization_config_repositories.py]: i think we normally do this by creating a custom feature flag and then adding it here: https://github.com/getsentry/getsentry/blob/master/getsentry/features.py#L232

File: src/social_auth/backends/bitbucket.py (hunk 1 above)
Comment (macqueen) [inline src/social_auth/backends/bitbucket.py]: can you test this to make sure? i think the only functionality you'd need to test is the bitbucket issue plugin

**Review by macqueen — CHANGES_REQUESTED** [review]

File: src/sentry/api/endpoints/organization_config_repositories.py (hunk 2 above)
Comment (MaxBittker) [inline src/sentry/api/endpoints/organization_config_repositories.py]: what would the difference be? letting us toggle it per-org?

Comment (MaxBittker) [inline src/sentry/api/endpoints/organization_config_repositories.py]: i don't think this needs to be early adopter-d for very long hopefully

Comment (macqueen) [inline src/sentry/api/endpoints/organization_config_repositories.py]: it just makes it clearer what the feature is and follows a pattern we always use. it will make this line be something like `if features.has('bitbucket-repos')` which is a lot nicer and easier to search for / remove when we need to.

File: src/sentry/api/endpoints/organization_config_repositories.py [hunk 3]
```diff
@@ -13,11 +14,12 @@ def get(self, request, organization):
         providers = []
         for provider_id in provider_bindings:
             provider = provider_bindings.get(provider_id)(id=provider_id)
-            providers.append({
-                'id': provider_id,
-                'name': provider.name,
-                'config': provider.get_config(),
-            })
+            if provider.name != 'Bitbucket' or features.has('organizations:bitbucket-repos'):
```
Comment (macqueen) [inline src/sentry/api/endpoints/organization_config_repositories.py]: nit: i think it's slightly nicer to use `provider_id != 'bitbucket'` instead of name

File: src/sentry/features/__init__.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
 default_manager.add('organizations:repos', OrganizationFeature)  # NOQA
 default_manager.add('organizations:release-commits', OrganizationFeature)  # NOQA
 default_manager.add('organizations:group-unmerge', OrganizationFeature)  # NOQA
+default_manager.add('organizations:bitbucket-repos')  # NOQA
```
Comment (macqueen) [inline src/sentry/features/__init__.py]: umm i think you want the `OrganizationFeature`, right?

File: src/sentry/static/sentry/app/views/stream/searchBar.jsx [hunk 5]
```diff
@@ -367,8 +367,8 @@ const SearchBar = React.createClass({
 
       searchItems[state.activeSearchItem].active = true;
       this.setState({searchItems: searchItems.slice(0)});
-    } else if (evt.key === 'Tab' && !this.isDefaultDropdown()) {
-      evt.preventDefault();
+    } else if ((evt.key === 'Tab' || evt.key === 'Enter') && !this.isDefaultDropdown()) {
```
Comment (macqueen) [inline src/sentry/static/sentry/app/views/stream/searchBar.jsx]: ummm, should these changes be in your pr? seems unrelated

**Review by macqueen — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/views/stream/searchBar.jsx (hunk 5 above)
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/stream/searchBar.jsx]: oh jeeze
good catch!

**Review by macqueen — APPROVED** [review]
